app/controllers/tracks/mentor_requests_controller.rb

```diff
@@ -3,10 +3,15 @@ class Tracks::MentorRequestsController < ApplicationController
   before_action :use_solution
 
   def new
+    unless Track::SearchSolutionsForMentoring.(current_user, @track, paginated: false).include?(@solution)
+      redirect_to track_path(@track)
+    end
```

Comment:
This can just be `return redirect_to track_path(@track) if @exercise.tutorial?` I think.

Reply:
This also checks if an exercise is started. I know it really won't happen technnically, but it's nice to guard using the `mentoring_unlocked?` method.

Reply:
Tests fail! I'm just going to use what you suggested :)